src/nectarchain/dqm/charge_integration.py

```diff
@@ -1,303 +1,401 @@
+from dqm_summary_processor import dqm_summary
+from matplotlib import pyplot as plt
+import numpy as np
+from ctapipe.visualization import CameraDisplay
+from ctapipe.instrument import CameraGeometry
+from traitlets.config.loader import Config
+import ctapipe.instrument.camera.readout
+from ctapipe.image import LocalPeakWindowSum
 
-from dqm_summary_processor import *
 
 class ChargeIntegration_HighLowGain(dqm_summary):
 
     def __init__(self, gaink):
 
-    	self.k = gaink
-    	return None
+        self.k = gaink
+        return None
 
-    def ConfigureForRun(self,path, Chan, Samp, Reader1):
-        #define number of channels and samples
+    def ConfigureForRun(self, path, Chan, Samp, Reader1):
+        # define number of channels and samples
         self.Chan = Chan
-        self.Samp= Samp
+        self.Samp = Samp
 
         self.counter_evt = 0
         self.counter_ped = 0
 
         self.camera = CameraGeometry.from_name("NectarCam-003")
         self.cmap = 'gnuplot2'
 
-        #reader1=EventSource(input_url=path, max_events = 1)
+        # reader1=EventSource(input_url=path, max_events = 1)
         self.subarray = Reader1.subarray
         subarray = Reader1.subarray
-        subarray.tel[0].camera.readout = ctapipe.instrument.camera.readout.CameraReadout.from_name("NectarCam")
-        config = Config({"LocalPeakWindowSum": {"window_shift": 12, "window_width": 12}})
+        subarray.tel[0].camera.readout = \
+            ctapipe.instrument.camera.readout.CameraReadout.from_name("NectarCam")
+        config = Config(
+            {"LocalPeakWindowSum": {"window_shift": 12, "window_width": 12}}
```

Comment:
I think there is actually a bug in the initial code here: the charge is integrated within 12 ns starting 12 ns before the peak! The extracted charge is thus not centered nor does it properly include the signal pulse.
@hashkar , could you have a look, please ?

Reply:
See #43